File: backend/app/schemas/vendor.py
from pydantic import BaseModel, EmailStr
from typing import Optional

class VendorCreate(BaseModel):
    name: str
    service_type: str
    phone: str
    email: EmailStr
    availability: Optional[str] = "Available"

class VendorResponse(BaseModel):
    id: int
    name: str
    service_type: str
    phone: str
    email: str
    availability: str

    class Config:
        from_attributes = True

File: frontend/src/types/vendor.ts
export interface Vendor {
  id: number;
  name: string;
  service_type: string;
  phone: string;
  email: string;
  availability: string;
}

export interface VendorCreate {
  name: string;
  service_type: string;
  phone: string;
  email: string;
  availability?: string;
}